packages/suite/src/components/suite/graph/TransactionsGraph/TransactionsGraph.tsx

```diff
@@ -90,6 +91,76 @@ export interface FiatGraphProps extends CommonProps {
 
 export type TransactionsGraphProps = CryptoGraphProps | FiatGraphProps;
 
+const emptyList: ReturnType<typeof selectAccountTransactionsWithNulls>[] = [];
+const useGraphUpdater = ({
+    shouldUpdateGraph,
+    account,
+}: {
+    shouldUpdateGraph: (abortController: AbortController) => Promise<any> | undefined;
+    account: Account | undefined;
+}) => {
+    const [currentPromise, setCurrentPromise] = useState<{
+        promiseId: string;
+        txIds: string[];
+        promise: Promise<any>;
+        abortController: AbortController;
+    } | null>(null);
+
+    const allTransactions = useSelector(state =>
+        account ? selectAccountTransactionsWithNulls(state, account.key) : emptyList,
+    );
+
+    const newestTransactions = allTransactions
+        .slice(0, 3)
+        .flatMap(tx => tx)
+        .filter((tx): tx is WalletAccountTransaction =>
+            Boolean(Boolean(tx) && tx && !isPending(tx)),
+        );
+
+    const txIds = newestTransactions.map(tx => tx.txid);
+
+    const promiseId = txIds.join('-');
+
+    useEffect(() => {
+        if (promiseId !== currentPromise?.promiseId && account) {
```

Comment:
It is sad that I cannot omit the graph update when `currentPromise` is null (on mount). Coz in that case, there is still no code that would update the graph data.

I like this , coz the refresh of the graph data and the graph itself are colocated.